Duckling/Time/KA/Rules.hs

```diff
@@ -38,36 +38,65 @@ import qualified Duckling.Ordinal.Types as TOrdinal
 import qualified Duckling.Time.Types as TTime
 import qualified Duckling.TimeGrain.Types as TG
 
+-- Georgian weekend
+
+geoWeekend :: TimeData
+geoWeekend = interval' TTime.Open (fri, mon)
+  where
+    fri = dayOfWeek 6
+    mon = dayOfWeek 7
+
+-- 
+
 ruleIntersect :: Rule
 ruleIntersect = Rule
   { name = "intersect"
   , pattern =
-    [ Predicate isNotLatent
+    [ Predicate $ or . sequence [isNotLatent, isGrainOfTime TG.Year]
     , Predicate $ or . sequence [isNotLatent, isGrainOfTime TG.Year]
     ]
-  , prod = \case
-      (Token Time td1:Token Time td2:_) ->
-        Token Time . notLatent <$> intersect td1 td2
-      _ -> Nothing
+  , prod = \tokens -> case tokens of
+    (Token Time td1:Token Time td2:_)
+      | (not $ TTime.latent td1) || (not $ TTime.latent td2) ->
+      Token Time . notLatent <$> intersect td1 td2
+    _ -> Nothing
   }
 
-ruleAbsorbInMonthYear :: Rule
-ruleAbsorbInMonthYear = Rule
-  { name = "in <named-month>|year"
+ruleAbsorbInYear :: Rule
+ruleAbsorbInYear = Rule
+  { name = "in year"
   , pattern =
-    [ Predicate $ or . sequence [isAMonth, isGrainOfTime TG.Year]
-    , regex " ?წელს| ?წელი| ?წლის| ?თვეს| ?თვის| ?თვის|-ში|ში"
```

Comment:
We changed this rule so that it only works on year. Changes in intersection rule (which were based on purely characteristics of Georgian language) makes it possible for some rules to be comprehended by other more basic rules.